### PR TITLE
Monorepo refactor

### PR DESCRIPTION
**Pull request by vlasin**

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -1,6 +1,10 @@
 index-state: 2023-10-29T16:11:26Z
 
-packages: .
+packages:
+    symbolic-base
+    symbolic-cardano
+    symbolic-examples
```
Comment (TurtlePU) [inline cabal.project:6]: Do `symbolic-examples` have to be a separate package from `symbolic-base`? I'd prefer to keep them as part of `base` to have the compiler benchmark closer to the `ArithmeticCircuit` definition

File: .github/workflows/main-push.yml [hunk 2]
```diff
@@ -37,14 +37,14 @@ jobs:
...
-        cabal new-build all -f Pedantic
+        cabal build all -f Pedantic
 
     - name: Upload package
       env:
         HACKAGE_PASSWORD: ${{ secrets.HACKAGE_PASSWORD }}
       shell: bash
       run: |
-        cabal sdist
+        cabal sdist all
```
Comment (TurtlePU) [inline .github/workflows/main-push.yml:47]: So now we have 4 package candidates on Hackage, right?

File: cabal.project (hunk 1 above)
Comment (vlasin) [reply cabal.project:6]: We currently have one example from `symbolic-cardano`.

File: .github/workflows/main-push.yml (hunk 2 above)
Comment (vlasin) [reply .github/workflows/main-push.yml:47]: Yes, possibly we should only upload the `symbolic-base` one.

File: cabal.project (hunk 1 above)
Comment (echatav) [reply cabal.project:6]: The `symbolic-examples` package and `symbolic-base` package appear to have duplicates between them in `symbolic-base/examples/` and `symbolic-examples/src`.

**Comment by echatav** [comment]
+1 monorepo

File: cabal.project (hunk 1 above)
Comment (vlasin) [reply cabal.project:6]: Thanks, fixed.

File: .github/workflows/main-push.yml (hunk 2 above)
Comment (vlasin) [reply .github/workflows/main-push.yml]: Changed to upload only `symbolic-base` now.

**Comment by vlasin** [comment]
@TurtlePU @echatav Any more comments? If not, please approve :)

**Review by TurtlePU — APPROVED** [review]

**Pull request merged**